### PR TITLE
Make server side network adaptation the default when creating `VideoPriorityBasedPolicy`

### PR DESCRIPTION
**Pull request by hensmi-amazon**
**Issue #:** None

**Description of changes:** We are no longer iterating on the priority policy without server side network adaptation. So it makes sense that it is the default.

**Testing:**
Confirmed that SSNA is used when not configured.

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
N/A no behavior is changed.

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
y

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
n

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
n

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/browser/app/meetingV2/video/PaginationManager.ts [hunk 1]
```diff
@@ -20,7 +20,9 @@ export default class PaginationManager<Type> {
   }
 
   remove(toRemove: Type) {
-    this.all.splice(this.all.indexOf(toRemove));
+    if (this.all.includes(toRemove)) {
```
Comment (hensmi-amazon) [inline demos/browser/app/meetingV2/video/PaginationManager.ts:23]: Some unrelated pagination fixes here

File: src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts [hunk 2]
```diff
@@ -3,15 +3,30 @@
...
 import VideoStreamIndex from '../videostreamindex/VideoStreamIndex';
 import VideoPreference from './VideoPreference';
 import { VideoPreferences } from './VideoPreferences';
 import VideoPriorityBasedPolicy from './VideoPriorityBasedPolicy';
+import VideoPriorityBasedPolicyConfig from './VideoPriorityBasedPolicyConfig';
 
+/** [[VideoAdaptiveProbePolicy]] wraps [[VideoPriorityBasedPolicy]] with customized behavior to automatically
+ * assign a high preference to content share.
+ *
+ * @deprecated This class is not compatible with latest priority policy features (i.e. server side network adaptation),
```
Comment (devalevenkatesh) [inline src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts:17]: 1. Since we are trying to help the builder with the documenation, probably a link to follow when we say "can be done trivially by the application" would be helpful.
2. This seems like a good detail for changelog entry as a build can understand the change better.

File: src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts [hunk 3]
```diff
@@ -3,15 +3,30 @@
...
+ * [[VideoPriorityBasedPolicy]] should be used instead. If high priority for content is desired it can be done trivially
+ * by the application.
+ */
 export default class VideoAdaptiveProbePolicy extends VideoPriorityBasedPolicy {
+  private static createConfig(): VideoPriorityBasedPolicyConfig {
```
Comment (devalevenkatesh) [inline src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts:22]: Hmm why static? Do we need to make this singleton for multiple `VideoAdaptiveProbePolicy` objects? If so checking if this approach is useful or do we need explicit object checking to preserve a singleton use of the config.

File: src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicyConfig.ts [hunk 4]
```diff
@@ -26,10 +26,11 @@ export default class VideoPriorityBasedPolicyConfig {
   private referenceBitrate: number = 0;
 
   /**
-   * Additional server side features to be enabled for network adaption. This
-   * may be overridden by the server. The default may be changed in future releases.
+   * Additional server side features to be enabled for network adaption. The only currently
+   * recommended value is `ServerSideNetworkAdaption.BandwidthProbingAndRemoteVideoQualityAdaption`
    */
-  serverSideNetworkAdaption = ServerSideNetworkAdaption.Default;
+  serverSideNetworkAdaption =
```
Comment (devalevenkatesh) [inline src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicyConfig.ts:32]: Just checking if this is a backward compatible change from builder APIs + behavior perspective and whether we need something to revert this back at runtime if builders face issues?

File: src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts (hunk 3 above)
Comment (hensmi-amazon) [reply src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts:22]: I'll add a comment. This is just a workaround for the silly typescript rule that `super` calls must be first in constructor. We don't need singleton usage (even though i'm not sure what customer would be creating multiple policies lol).

File: src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicyConfig.ts (hunk 4 above)
Comment (hensmi-amazon) [reply src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicyConfig.ts:32]: TBH I don't think many customers use priority policy without SSNA. Unfortunately since it is client side we don't actually know the number, but given that we don't get any complaints about it (and there is plenty to complain about) I think the number isn't high (and they probably aren't updating the SDK quickly anyways). Customers can of course easily switch to None if they are happy with what they got, but TBH I want to eventually remove the ability to set None (since it isn't technically a change in behavior, I don't see much issue with this).

That being said, I do technically have that knob on the backend, but it's not really a button i want to press given the existing policy has a lot of issues.

File: src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts (hunk 2 above)
Comment (hensmi-amazon) [reply src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts:17]: Will update!

File: src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts (hunk 3 above)
Comment (hensmi-amazon) [reply src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts:22]: Actually i misunderstood comments i wrote before. We don't need to deprecate this class, my bad.

File: src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts (hunk 2 above)
Comment (hensmi-amazon) [reply src/videodownlinkbandwidthpolicy/VideoAdaptiveProbePolicy.ts:17]: actually i misunderstood a comment i wrote before/how this is used (when simulcast is enabled but a policy is not set). I don't need to deprecate this yet.

File: src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicyConfig.ts (hunk 4 above)
Comment (devalevenkatesh) [reply src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicyConfig.ts:32]: Cool I am fine with this approach, probably we can leverage client events to help us here in future.

**Review by devalevenkatesh — DISMISSED** [review]

File: demos/browser/app/meetingV2/video/VideoTileCollection.ts [hunk 5]
```diff
@@ -452,10 +454,10 @@ export default class VideoTileCollection implements AudioVideoObserver {
     // We need to manually control visibility of paused tiles anyways so we just do
     // everything here, even though the preference manager adding/removing will
     // result in tile callbacks as well.
-    for (let [index, videoTile] of this.tileIndexToDemoVideoTile.entries()) {
+    for (let videoTile of this.tileIndexToDemoVideoTile.values()) {
         if (attendeesToShow.includes(videoTile.attendeeId)) {
             videoTile.show(false);
-        } else if (this.tileIndexToTileId[index] !== this.findContentTileId()) { // Always show content
```
Comment (hensmi-amazon) [inline demos/browser/app/meetingV2/video/VideoTileCollection.ts]: This breaks because of the local content that automatically makes a tile. It doesn't really need to make UX decisions like this (not like UX eng ever assisted with demo) so simpler is better IMO
